panther_hardware_interfaces/src/roboteq_driver.cpp

```diff
@@ -0,0 +1,345 @@
+  }
+}
+
+RoboteqDriverFeedback RoboteqDriver::ReadRoboteqDriverFeedback()
+{
+  RoboteqDriverFeedback fb;
+
+  // uint32_t
+  // already does locking when accessing rpdo
+  fb.motor_1.pos = rpdo_mapped[0x2106][1];
+  fb.motor_2.pos = rpdo_mapped[0x2106][2];
+
+  fb.motor_1.vel = rpdo_mapped[0x2106][3];
+  fb.motor_2.vel = rpdo_mapped[0x2106][4];
+
+  fb.motor_1.current = rpdo_mapped[0x2106][5];
+  fb.motor_2.current = rpdo_mapped[0x2106][6];
+
+  fb.fault_flags = GetByte(rpdo_mapped[0x2106][7], 0);
+  fb.script_flags = GetByte(rpdo_mapped[0x2106][7], 2);
+
+  fb.runtime_stat_flag_motor_1 = GetByte(rpdo_mapped[0x2106][8], 0);
+  fb.runtime_stat_flag_motor_2 = GetByte(rpdo_mapped[0x2106][8], 1);
+
+  std::unique_lock<std::mutex> lck(rpdo_timestamp_mtx_);
+  fb.timestamp = last_rpdo_write_timestamp_;
+
+  return fb;
+}
+
+// todo check what happens when publishing is stopped (on hold - waiting for decision on changing to
+// PDO)
+void RoboteqDriver::SendRoboteqCmdChannel1(int32_t cmd)
+{
+  try {
+    SyncSdoWrite<int32_t>(0x2000, 1, cmd);
+  } catch (const std::runtime_error & e) {
+    throw std::runtime_error(
+      "Error when trying to send channel 1 Roboteq command: " + std::string(e.what()));
+  }
+}
+
+void RoboteqDriver::SendRoboteqCmdChannel2(int32_t cmd)
+{
+  try {
+    SyncSdoWrite<int32_t>(0x2000, 2, cmd);
+  } catch (const std::runtime_error & e) {
+    throw std::runtime_error(
+      "Error when trying to send channel 2 Roboteq command: " + std::string(e.what()));
+  }
+}
+
+void RoboteqDriver::ResetRoboteqScript()
+{
+  try {
+    SyncSdoWrite<uint8_t>(0x2018, 0, 2);
+  } catch (const std::runtime_error & e) {
+    throw std::runtime_error("Error when trying to reset Roboteq script: " + std::string(e.what()));
+  }
+}
+
+void RoboteqDriver::TurnOnEstop()
+{
+  // Cmd_ESTOP
+  try {
+    SyncSdoWrite<uint8_t>(0x200C, 0, 1);
+  } catch (const std::runtime_error & e) {
+    throw std::runtime_error("Error when trying to turn on estop: " + std::string(e.what()));
+  }
+}
+
+void RoboteqDriver::TurnOffEstop()
+{
+  // Cmd_MGO
+  try {
+    SyncSdoWrite<uint8_t>(0x200D, 0, 1);
+  } catch (const std::runtime_error & e) {
+    throw std::runtime_error("Error when trying to turn off estop: " + std::string(e.what()));
+  }
+}
+
+void RoboteqDriver::TurnOnSafetyStopChannel1()
+{
+  // Cmd_SFT Safety Stop
+  try {
+    SyncSdoWrite<uint8_t>(0x202C, 0, 1);
+  } catch (const std::runtime_error & e) {
+    throw std::runtime_error(
+      "Error when trying to turn on safety stop on channel 1: " + std::string(e.what()));
+  }
+}
+
+void RoboteqDriver::TurnOnSafetyStopChannel2()
+{
+  // Cmd_SFT Safety Stop
+  try {
+    SyncSdoWrite<uint8_t>(0x202C, 0, 2);
+  } catch (const std::runtime_error & e) {
+    throw std::runtime_error(
+      "Error when trying to turn on safety stop on channel 2: " + std::string(e.what()));
+  }
+}
+
+template <typename type>
```

Comment:
```suggestion
template <typename T>
```
This way are we using this?